src/carousel.tsx

```diff
@@ -0,0 +1,727 @@
+                .reverse()
+                .find((pos) => pos - scrollLeft < -1);
+            carouselEl.scrollTo({left: prevPageScrollPosition, behavior: 'smooth'});
+        }
+    }, [scrollPositions]);
+
+    const goNext = React.useCallback(() => {
+        const carouselEl = carouselRef.current;
+        if (carouselEl) {
+            const {scrollLeft} = carouselEl;
+            const nextPageScrollPosition = scrollPositions.find((pos) => pos - scrollLeft > 1);
+            carouselEl.scrollTo({left: nextPageScrollPosition, behavior: 'smooth'});
+        }
+    }, [scrollPositions]);
+
+    React.useEffect(() => {
+        if (autoplay) {
+            const time = typeof autoplay === 'boolean' ? 5000 : autoplay.time;
+            const loop = typeof autoplay === 'object' && autoplay.loop;
+            const interval = setInterval(() => {
+                if (scrollRight !== 0) {
+                    goNext();
+                } else if (loop) {
+                    carouselRef.current?.scrollTo({left: 0, behavior: 'smooth'});
+                }
+            }, time);
+            return () => clearInterval(interval);
+        }
+    }, [autoplay, goNext, scrollRight]);
+
+    const currentPageIndex = calcCurrentPageIndex(scrollLeft, pagesScrollPositions);
+
+    React.useEffect(() => {
+        if (onPageChange) {
+            const lastShownItemIndex = Math.min(
+                (currentPageIndex + 1) * itemsPerPageFloor - 1,
+                items.length - 1
+            );
+            const shownItemIndexes = [];
+            for (let i = 0; i < itemsPerPageFloor; i++) {
+                shownItemIndexes.unshift(lastShownItemIndex - i);
+            }
+            onPageChange({pageIndex: currentPageIndex, shownItemIndexes});
+        }
+    }, [currentPageIndex, items.length, itemsPerPageFloor, onPageChange]);
+
+    let bullets: React.ReactNode = null;
+
+    if (renderBullets) {
+        bullets = renderBullets({numPages: pagesCount, currentIndex: currentPageIndex, onPress: goToPage});
+    } else if (withBullets) {
+        bullets = <PageBullets numPages={pagesCount} currentIndex={currentPageIndex} onPress={goToPage} />;
+    }
+
+    return (
+        <Stack space={24} dataAttributes={dataAttributes}>
+            <div className={classes.carouselContainer}>
+                <ThemeVariant isInverse={false}>
+                    <Touchable
+                        className={classNames(classes.arrowButton, 'prev')}
+                        aria-label={texts.carouselPrevButton}
+                        onPress={goPrev}
+                        disabled={!showPrevArrow}
+                    >
+                        <IconChevronLeftRegular />
+                    </Touchable>
+                </ThemeVariant>
+                <div className={classNames(classes.carousel, {centered})} ref={carouselRef}>
+                    {items.map((item, index) => (
+                        <div key={index} className={classes.slide} data-item>
```

Comment:
`data-slide` ? maybe is more precise and less generic that `item`

Reply:
disagree. What's a slide? a page? a carousel page can have multiple carousel items. It's quite clear to me what a page is and what a an item is, but a slide is confusing to me, it could be synonym of item or synonym of page. Also, I think I'm going to rename the slide clasname to item too